### PR TITLE
Add example RESTCONF DOM actions app

### PR DESCRIPTION
**Pull request by ihrasko**
Add an example application which shows how
to register example DOM action to be available
for invocation via RESTCONF interface.

Signed-off-by: Ivan Hrasko <ivan.hrasko@pantheon.tech>

### CODE REVIEW

File: lighty-examples/lighty-community-restconf-dom-actions-app/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+![architecture](docs/restconf-netconf-controller-architecture.svg)
+
+This roughly translates to OpenDaylight feature set installed by karaf command:
+```
+feature:install odl-netconf-all
+```
+
+## Build and Run
+build the project: ```mvn clean install```
```
Comment (vainyksi) [inline lighty-examples/lighty-community-restconf-dom-actions-app/README.md:18]: -> `build the root project`

File: lighty-examples/lighty-community-restconf-dom-actions-app/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+2. You will receive output like the following:
+
+```json
+{
+    "example-data-center:output": {
+        "example-data-center:start-finished-at": "2021-09-09T16:20:00Z"
+    }
+}
+```
```
Comment (vainyksi) [inline lighty-examples/lighty-community-restconf-dom-actions-app/README.md:51]: please describe briefly what is happening here, what represents the payload and from where the output came from (lighty pass-through the input value prepared in `DeviceStartActionImpl`)

File: lighty-examples/lighty-community-restconf-dom-actions-app/src/main/java/io/lighty/examples/controllers/domactions/DOMActionRegistrationUtil.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import org.opendaylight.mdsal.dom.api.DOMDataTreeIdentifier;
+import org.opendaylight.yang.gen.v1.urn.example.data.center.rev180807.Device;
+import org.opendaylight.yang.gen.v1.urn.example.data.center.rev180807.device.Start;
+import org.opendaylight.yangtools.concepts.ObjectRegistration;
+import org.opendaylight.yangtools.yang.data.api.YangInstanceIdentifier;
+import org.opendaylight.yangtools.yang.model.api.stmt.SchemaNodeIdentifier;
+
+final class DOMActionRegistrationUtil {
+    private DOMActionRegistrationUtil() {
+        // hidden on purpose
```
Comment (vainyksi) [inline lighty-examples/lighty-community-restconf-dom-actions-app/src/main/java/io/lighty/examples/controllers/domactions/DOMActionRegistrationUtil.java:24]: mention the reason: `utility class`

File: lighty-examples/lighty-community-restconf-dom-actions-app/src/main/resources/log4j.properties [hunk 4]
```diff
@@ -0,0 +1,4 @@
+log4j.rootLogger=INFO, STDOUT
```
Comment (vainyksi) [inline lighty-examples/lighty-community-restconf-dom-actions-app/src/main/resources/log4j.properties:1]: log4j config should not be needed
the config is inherited from `log4j-properties` dependency

**Review by vainyksi — CHANGES_REQUESTED** [review]

File: lighty-examples/lighty-community-restconf-dom-actions-app/pom.xml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  terms of the Eclipse Public License v1.0 which accompanies this distribution,
+  and is available at https://www.eclipse.org/legal/epl-v10.html
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>io.lighty.core</groupId>
+        <artifactId>lighty-app-parent</artifactId>
+        <version>14.0.1-SNAPSHOT</version>
```
Comment (vainyksi) [inline lighty-examples/lighty-community-restconf-dom-actions-app/pom.xml:15]: version should be `14.1.0-SNAPSHOT`

**Review by vainyksi — CHANGES_REQUESTED** [review]

File: lighty-examples/lighty-community-restconf-dom-actions-app/src/main/java/io/lighty/examples/controllers/domactions/DOMActionRegistrationUtil.java (hunk 3 above)
Comment (ihrasko) [reply lighty-examples/lighty-community-restconf-dom-actions-app/src/main/java/io/lighty/examples/controllers/domactions/DOMActionRegistrationUtil.java]: done

File: lighty-examples/lighty-community-restconf-dom-actions-app/README.md (hunk 1 above)
Comment (ihrasko) [reply lighty-examples/lighty-community-restconf-dom-actions-app/README.md]: done

File: lighty-examples/lighty-community-restconf-dom-actions-app/src/main/resources/log4j.properties (hunk 4 above)
Comment (ihrasko) [reply lighty-examples/lighty-community-restconf-dom-actions-app/src/main/resources/log4j.properties]: removed

File: lighty-examples/lighty-community-restconf-dom-actions-app/pom.xml (hunk 5 above)
Comment (ihrasko) [reply lighty-examples/lighty-community-restconf-dom-actions-app/pom.xml]: It should be 14.1.1-SNAPSHOT according to base branch you have choose.

Comment (vainyksi) [reply lighty-examples/lighty-community-restconf-dom-actions-app/pom.xml]: thanks for your correction

**Review by vainyksi — APPROVED** [review]

**Pull request merged**